src/ann_solo/reader.py

```diff
@@ -2,21 +2,26 @@
 import logging
 import os
 import pickle
+import gzip
+import zipfile
 from functools import lru_cache
 from typing import Iterator
 from typing import List
-from typing import Tuple
+from typing import Tuple, Dict, IO, Iterator, Sequence, Union
+import io
 
 import joblib
 import numpy as np
 import pandas as pd
 import tqdm
-from pyteomics import mgf
+from lxml.etree import LxmlError
+from pyteomics import mgf, mzml, mzxml
 from spectrum_utils.spectrum import MsmsSpectrum
+import lzma
```

Comment:
**nitpick:** This is an import from the standard library, and should be included in the first block.